django_filters/filterset.py

```diff
@@ -376,7 +377,9 @@ def filter_for_field(cls, field, field_name, lookup_expr='exact'):
             "#customise-filter-generation-with-filter-overrides"
         ) % (cls.__name__, field_name, lookup_expr, field.__class__.__name__)
 
-        return filter_class(**default)
+        f = filter_class(**default)
+        f.model = model
+        return f
```

Comment:
I'm not sure about this change. On the one hand, I understand that it improves your use case. On the other, it works against how/when the parent and model are bound to the filter, which is done during filterset initialization, not class creation. 

I'm wondering if the DRF backend should instantiate the fitlerset when generating the schema, instead of using the class. That said, I'm not sure if that's possible during schema generation? @carltongibson ?